BeVolt/BSP/Simulator/Src/BSP_Strobelight.c

```diff
@@ -0,0 +1,43 @@
+#include "../../Inc/BSP_Strobelight.h"
+#include <stdint.h>
+#include <stdlib.h>
+#define OFF 0
+#define PULSING 1
+
+static const char* file = "BSP/Simulator/DataGeneration/Data/Strobe.csv";
+int StrobeStatus = 0;
+
+/**
+ * @brief   Create text file if none exists (Simulator)
+ * @param   None
+ * @return  None
+ */
+void BSP_Strobe_Init(void) {
+	FILE* fp = fopen(file, "w+");
+	fprintf(fp, "0");
+	fclose(fp);
+}
+
+/**
+ * @brief   Turns on Strobe (pulse once per second)
+ * @param   None
+ * @return  None
+ */
+void BSP_Strobe_On(void) {
+    StrobeStatus = PULSING;    //1 = pulsing
+    FILE* fp = fopen(file, "w");
+    fprintf(fp, "%d", StrobeStatus);
+    fclose(fp);
+}
+
+/**
+ * @brief   Turns off Strobe
+ * @param   None
+ * @return  None
+ */
+void BSP_Strobe_Off(void) {
+    StrobeStatus = OFF;     //0 = off
+    FILE* fp = fopen(file, "w");
+    fprintf(fp, "%d", StrobeStatus);
+    fclose(fp);
+}
```

Comment:
Each time you open the file, add a lock to the file. Check out some of the other simulation branches to see how that works.

Reply:
I've looked at the other simulation branches, I have no idea what "adding a lock" is though..

Reply:
Locking the file is just done like this:

After you open a file, put the following lines after
```
int fno = fileno(fp);
flock(fno, LOCK_EX);
```
And then before you close the file:
```
flock(fno, LOCK_UN);
```
And you'll have to include this header: ```<sys/file.h>```